### PR TITLE
Add GitHub Actions workflow to Rails

### PR DESCRIPTION
**Pull request by eileencodes**
This PR adds the necessary workflow to use GitHub Actions on the
Rails repo. There are a couple of things missing but the majority of
tests are passing for Linux. :tada:

The changes to ActionPack and Railties are due to other CI's pretending
they are a tty, but Actions doesn't act as a tty so we need to handle
these differently.

Co-Authored-By: John Hawthorn <john@hawthorn.email>
Co-Authored-By: Aaron Patterson <aaron.patterson@gmail.com>
Co-Authored-By: Edward Thomson <ethomson@edwardthomson.com>

### CODE REVIEW

**Comment by rafaelfranca** [comment]
What advantages does it brings over buildkite? Can we parallelize builds across multiple machines easily?

**Comment by eileencodes** [comment]
We don't yet have everything we'd need to switch over. I wanted to get this running so we can see how it feels, figure out the pros/cons, and potentially make our CI a bit more robust with windows and/or macos testing. I've opened this as a starting point for discussion. 😄 We can always start on a smaller project under the Rails name as well.

**Comment by rafaelfranca** [comment]
:+1: I do think without a way to parallelize tests between machines we will slow down our suite but running it on mac os and windows and keeping buildkite to run our linux tests would be great.

**Comment by yahonda** [comment]
Would you consider to implement RuboCop check for pull requests?  We may not have to wait for CodeClimate releases newer version of channels like `rubocop-0-74`.

File: .github/workflows/workflow.yml [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+      run: |
+        sudo apt-get update
+        sudo apt-get install -y bison ruby ruby-dev libffi-dev libgdbm-dev libgmp-dev libjemalloc-dev libncurses5-dev libncursesw5-dev libreadline6-dev libssl-dev libyaml-dev openssl zlib1g-dev libmysqlclient-dev libpq-dev sqlite3 libsqlite3-dev redis-tools curl libxml2-dev tzdata
+        curl -sL https://deb.nodesource.com/setup_10.x | sudo -E bash -
+        sudo apt-get install -y nodejs
+        sudo gem install bundler -v '1.17.3'
+    - name: Install dependencies
+      run: |
+        sudo gem install bundler -v '1.17.3'
+        bundle install -j8
```
Comment (morgoth) [inline .github/workflows/workflow.yml]: @eileencodes Do you know what would be the solution to cache gems between builds using GH actions? If I'm not wrong every build has to fetch all the gems even when there were no changes to Gemfile.lock

Comment (ethomson) [reply .github/workflows/workflow.yml]: We're working on adding caching to GitHub Actions - we should have it available soon, in the meantime you're right that this is fetching all gems.

Comment (rmacklin) [reply .github/workflows/workflow.yml]: For anyone following along, caching dependencies is now supported:
https://github.blog/changelog/2019-11-04-github-actions-adds-dependency-caching/
🎉

**Pull request closed without merging**